definitions/features/hammer.rb

```diff
@@ -92,7 +92,8 @@ def on_invalid_host(custom_config)
   end
 
   def on_invalid_password(custom_config)
-    if !ready? && custom_config[:foreman][:password].nil?
+    username = custom_config[:foreman][:username]
+    if !ready? && custom_config[:foreman][:password].nil? && (username == username_from_answers)
```

Comment:
It seems there is error in the original condition. Here if `ready?` is false two things could have happened 1/ there was invalid password in hammer config 2/ There was NO admin password in hammer configs and invalid password in installer answers. This condition should execute the block only in case 1/. (For the 2/ we have already tried the password from answers). So my take on the condition would be: 
`if !ready? && custom_config[:foreman][:password] != password_from_answers`

For username we should test later to not lose the logging.